docs/UserGuide.md

```diff
@@ -4,16 +4,128 @@ title: "User Guide"
+    - [4.1.2 Editing a student: `edit`](#editing-a-student-edit)
+    - [4.1.3 Deleting a student: `deleteStu`](#deleting-a-student-deleteStu)
+    - [4.1.4 Listing students: `listStu`](#listing-students-listStu)
+    - [4.1.5 Find Students by Name: `find`](#find-students-find)
+- [4.2 Tutorial Data Related Commands](#tutorial-data-commands)
+    - [4.2.1 Adding a tutorial: `addTut`](#adding-a-tutorial-addTut)
+    - [4.2.2 Deleting a tutorial: `deleteTut`](#deleting-a-tutorial-deleteTut)
+    - [4.2.3 Listing tutorials: `listTut`](#listing-tutorials-listTut)
+- [4.3 Assignment Data Related Commands](#assignment-data-commands)
+    - [4.3.1 Adding an assignment: `addAsg`](#adding-an-assigment-addAsg)
+    - [4.3.2 Deleting an assignment: `deleteTut`](#deleting-an-assignment-deleteAsg)
+    - [4.3.3 Listing assignments: `listAsg`](#listing-assignments-listAsg)
+    - [4.3.4 Marking an assignment: `markAsg`](#marking-an-assignment-markAsg)
+    - [4.3.5 Unmarking an assignment: `unmarkAsg`](#unmarking-an-assignment-unmarkAsg)
+    - [4.3.6 Checking an assignment: `checkAsg`](#checking-an-assignment-checkAsg)
+- [4.4 Attendance Data Related Commands](#attendance-data-commands)
+  - [4.4.1 Marking an attendance: `attend`](#adding-an-assigment-addAsg)
+  - [4.4.2 Unmarking an attendance: `TBC`](#deleting-an-assignment-deleteAsg)
+- [4.5 General Commands](#general-commands)
+    - [4.5.1 Viewing a Student on the Student Card](#viewing-a-student-on-the-student-card)
+    - [4.5.2 Clearing all entries: `clear`](#clearing-all-entries-clear)
+    - [4.5.3 Exiting the program: `exit`](#exiting-the-program-exit)
+    - [4.5.4 Viewing help: `help`](#viewing-help-help)
+- [4.6 Saving the Data](#saving-the-data)
+- [4.7 Editing the Data File](#editing-the-data-file)
+
+[5. FAQ](#faq)
+
+[6. Known issues](#known-issues)
+
+[7. Glossary](#glossary)
+
+[8. Command summary](#command-summary)
+
+--------------------------------------------------------------------------------------------------------------------
+
+## 1. Introduction
+<a id="introduction"></a>
+
+### 1.1 What is TrackMate
+<a id="what-is-trackmate"></a>
+
+TO BE DONE
+
+### 1.2 User Proficiency and Expectations
+<a id="user-proficiency-and-expectations"></a>
+
+TO BE DONE
+
+### 1.3 Why This Guide Matters
+<a id="why-this-guide-matters"></a>
+
+TO BE DONE
 
 --------------------------------------------------------------------------------------------------------------------
 
-## Quick start
+## 2. How to use this User Guide
+<a id="how-to-use-this-user-guide"></a>
+
+This section is designed to help users effectively navigate the User Guide for TrackMate. Below, you'll find 
+information on how to interpret icons, formatting, and instructions provided throughout the document.
+
+### 2.1 Navigating the Document:
+
+TO BE DONE
+
+### 2.2 Sections:
+<a id="sections"></a>
+
+Below is a detailed overview of the main sections within this User Guide and what can be expected from each section.
```

Comment:
I like how you mark parts that yet to be done to 'TO BE DONE' so it is easier to be referenced to in the future.